cmd/prepare-storage.go

```diff
@@ -69,21 +78,36 @@ func formatXLMigrateLocalEndpoints(endpoints EndpointList) error {
 
 // Cleans up tmp directory of local disks.
 func formatXLCleanupTmpLocalEndpoints(endpoints EndpointList) error {
-	for _, endpoint := range endpoints {
+	g := errgroup.WithNErrs(len(endpoints))
+	for index, endpoint := range endpoints {
 		if !endpoint.IsLocal {
 			continue
 		}
-		formatPath := pathJoin(endpoint.Path, minioMetaBucket, formatConfigFile)
-		if _, err := os.Stat(formatPath); err != nil {
-			if os.IsNotExist(err) {
-				continue
+		index := index
+		g.Go(func() error {
+			epPath := endpoints[index].Path
+			formatPath := pathJoin(epPath, minioMetaBucket, formatConfigFile)
+			if _, err := os.Stat(formatPath); err != nil {
+				if os.IsNotExist(err) {
+					return nil
+				}
+				return err
 			}
-			return err
-		}
-		if err := removeAll(pathJoin(endpoint.Path, minioMetaTmpBucket)); err != nil {
-			return err
-		}
-		if err := mkdirAll(pathJoin(endpoint.Path, minioMetaTmpBucket), 0777); err != nil {
+			if _, err := os.Stat(pathJoin(epPath, minioMetaTmpBucket+"-old")); err != nil {
+				if os.IsNotExist(err) {
+					if err := renameAll(pathJoin(epPath, minioMetaTmpBucket),
+						pathJoin(epPath, minioMetaTmpBucket+"-old")); err != nil {
+						return err
+					}
+				}
+				// Removal of tmp-old folder is backgrounded completely.
+				go removeAll(pathJoin(epPath, minioMetaTmpBucket+"-old"))
```

Comment:
`removeAll` shouldn't be called when `os.Stat` returns errors other than `os.IsNotExist`, right?

Reply:
yes @krisis